clients/filesystem-fuse/src/gvfs_fileset_fs.rs

```diff
@@ -0,0 +1,135 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *  http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+use crate::config::AppConfig;
+use crate::filesystem::{FileStat, FileSystemCapacity, FileSystemContext, PathFileSystem, Result};
+use crate::gravitino_client::GravitinoClient;
+use crate::opened_file::{OpenFileFlags, OpenedFile};
+use async_trait::async_trait;
+use fuse3::Errno;
+use std::path::{Path, PathBuf};
+
+pub(crate) struct GravitinoFileSystemConfig {}
+
+/// GravitinoFileSystem is a filesystem that is associated with a fileset in Gravitino.
+/// It mapping the fileset path to the original data storage path. and delegate the operation
+/// to the inner filesystem like S3 GCS, JuiceFS.
+pub(crate) struct GvfsFilesetFs {
+    fs: Box<dyn PathFileSystem>,
```

Comment:
`fs` is too gneneral, use a better name?

Reply:
Is `inner_fs` ok?

Reply:
something like `physical_fs`?  the underlying fs are doing real physical operations

Reply:
fix